src/docfx/lib/git/NativeMethods.cs

```diff
@@ -0,0 +1,241 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Microsoft.Docs
+{
+    internal static class NativeMethods
```

Comment:
can we disable stylecop entirely for this file and Git.cs?

We should keep the function names and struct fields the same as LibGit2, like `git_object_id`